storage/fanout.go

```diff
@@ -502,6 +506,9 @@ func (c *genericMergeSeriesSet) Next() bool {
 			if set.Next() {
 				heap.Push(&c.heap, set)
 			}
+			if err := set.Err(); err != nil && c.err == nil {
```

Comment:
I think we can skip those, and just do return false in case of error and then allow for loop to loop through `set.Err` as we had before, just easier ..

Reply:
No a blocker here, up to you